docs/_articles/en/v3.x/salesforce-graph-engine/rules.md

```diff
@@ -139,6 +137,79 @@ Because this abstract class or interface has no implementations or extensions, i
 
 Because UnimplementedType rule excludes `global` scoped classes from consideration, these classes are prevented from being thrown as false positives and aren’t false negatives.
 
+## ApexNullPointerExceptionRule#
+
+ApexNullPointerExceptionRule identifies Apex operations that dereference null objects and throw NullPointerExceptions. NullPointerExceptions generally indicate underlying problems in your code to address. 
+
+Examples:
+
+```
+public void example1() {
+	Object myStr = null;
+	System.debug(myStr.toLowerCase()); // throws System.NullPointerException 
+// since method is invoked on null object.
+}
+
+
+public void example2(String myStr) {
+	if (myStr == null) {
+		System.debug(myStr.toLowerCase());
+// throws System.NullPointerException since the conditional
+// confirms that myStr is null when it reaches here.
+	}
+}
+
+
+public void example3() {
+	Integer i; // Not initialized
+	Integer y = i + 2; // throws System.NullPointerException since 
+// binary operation is invoked on null object
+}
+```
+
+In these examples, Apex Runtime throws a `System.NullPointerException` at the place where an operation is performed on a null object. The Graph Engine ApexNullPointerException rule preemptively identifies these operations. 
+
+To fix NullPointerException issues, use one of these methods.
+
+- Check that the object is `not-null` before performing the operation
+- Make sure that all variables are initialized
+
+Avoid initializing variables to `null`. If your logic demands initialization to `null`, make sure to reassign a value to your variable before you invoke an operation on it.
+
+To run the both data flow analysis rules–ApexFlsViolationRule and ApexNullPointerExceptionRule–run `scanner:run:dfa`. Alternatively, run `scanner:run:dfa --category “Error Prone”` to run ApexNullPointerExceptionRule only.
```

Comment:
Same comment around formatting as [before](https://github.com/forcedotcom/sfdx-scanner/pull/1034/files#r1148063206).
Can we use the term "Path-based rules" instead of "data flow analysis" rules?

Full command (we need to be careful about the quotes and make sure it's not angled. Mac OS has this autocorrect that constantly breaks command syntax):
```
sfdx scanner:run:dfa --category "Error Prone" --projectdir /project/dir --target /project/dir/target
```